keras_nlp/models/roberta.py

```diff
@@ -0,0 +1,302 @@
+        x = keras.layers.LayerNormalization(
+            name="embeddings_layer_norm",
+            axis=-1,
+            epsilon=1e-5,  # Original paper uses this epsilon value
+            dtype=tf.float32,
+        )(embedding)
+        x = keras.layers.Dropout(
+            dropout,
+            name="embeddings_dropout",
+        )(x)
+
+        # Apply successive transformer encoder blocks.
+        for i in range(num_layers):
+            x = TransformerEncoder(
+                num_heads=num_heads,
+                intermediate_dim=intermediate_dim,
+                activation=lambda x: keras.activations.gelu(
+                    x, approximate=True
+                ),
+                dropout=dropout,
+                kernel_initializer=_roberta_kernel_initializer(),
+                name=f"""transformer_layer_{i}""",
+            )(x, padding_mask=input_mask)
+
+        # Construct the two RoBerta outputs.
+        output = x
+
+        # Instantiate using Functional API Model constructor
+        super().__init__(
+            inputs={
+                "input_ids": token_id_input,
+                # "segment_ids": segment_id_input,
+                "input_mask": input_mask,
+            },
+            outputs=output,
+            name=name,
+            trainable=trainable,
+        )
+        # All references to `self` below this line
+        self.token_embedding = (
+            token_and_position_embedding_layer.token_embedding
+        )
+        self.vocabulary_size = vocabulary_size
+        self.hidden_dim = hidden_dim
+        self.intermediate_dim = intermediate_dim
+        self.num_layers = num_layers
+        self.num_heads = num_heads
+        self.max_sequence_length = max_sequence_length
+        self.intermediate_dim = intermediate_dim
+        self.dropout = dropout
+        # BOS token '<s>' is equivalent to '[CLS]' from BERT
+        self.bos_token_index = bos_token_index
+
+    def get_config(self):
+        config = super().get_config()
+        config.update(
+            {
+                "vocabulary_size": self.vocabulary_size,
+                "hidden_dim": self.hidden_dim,
+                "intermediate_dim": self.intermediate_dim,
+                "num_layers": self.num_layers,
+                "num_heads": self.num_heads,
+                "max_sequence_length": self.max_sequence_length,
+                "dropout": self.dropout,
+                "bos_token_index": self.bos_token_index,
+            }
+        )
+        return config
+
+
+# TODO: add RobertaMLM, different from BERT.
+
+
+class RobertaClassifier(keras.Model):
+    """Roberta encoder model with a classification head.
+
+    Args:
+        base_model: A `keras_nlp.models.Roberta` to encode inputs.
+        num_classes: Int. Number of classes to predict.
+        hidden_dim: Int. The size of the pooler layer.
+        bos_token_index: Int. Index of <s> token in the vocabulary. Equivalent
```

Comment:
this feels over-parameterized a bit. we have this on the base encoder and the classifier? if you are doing classification with a custom bos token index, are you expected to set one? or both?

maybe we can just read this param from the base encoder?